sssd_test_framework/utils/pam.py

```diff
@@ -0,0 +1,284 @@
+""""PAM Tools."""
+
+from __future__ import annotations
+
+from pytest_mh import MultihostHost, MultihostUtility
+from pytest_mh.ssh import SSHLog
+from pytest_mh.utils.fs import LinuxFileSystem
+
+__all__ = [
+    "PAMUtils",
+    "PAMAccess",
+    "PAMFaillock",
+]
+
+
+class PAMUtils(MultihostUtility[MultihostHost]):
+    """
+    Configuring various PAM modules
+    """
+
+    def __init__(self, host: MultihostHost, fs: LinuxFileSystem) -> None:
+        """
+        :param host: Remote host instance
+        :type host: MultihostHost
+        :param fs: Linux file system instance
+        :type fs: LinuxFileSystem
+        """
+        super().__init__(host)
+
+        self.fs: LinuxFileSystem = fs
+
+    def access(self, file: str = "/etc/security/access.conf") -> PAMAccess:
+        """
+        :param file: PAM Access file name.
+        :type file: str
+        :return: PAM Access object
+        :rtype: PAMAccess
+        """
+        return PAMAccess(self, file)
+
+    def faillock(self, file: str = "/etc/security/faillock.conf") -> PAMFaillock:
+        """
+        :param file: PAM Faillock file name.
+        :type file: str
+        :return: PAM Faillock object
+        :rtype: PAMFaillock
+        """
+        return PAMFaillock(self, file)
+
+
+class PAMAccess:
+    """
+    Management of PAM Access on the client host.
+
+    .. code-block:: python
+        :caption: Example usage
+
+        @pytest.mark.topology(KnownTopologyGroup.AnyProvider)
+        def test_example(client: Client, provider: GenericProvider):
+            # Add users
+            provider.user("user-1").add()
+            provider.user("user-2").add()
+
+            # Add rule to permit "user-1" and deny "user-2"
+            access = client.pam.access()
+            access.config_set([{"access": "+", "user": "user-1", "origin": "ALL"},
+                               {"access": "-", "user": "user-2", "origin": "ALL"}])
+            access.config_write()
+
+            client.sssd.authselect.enable_feature(["with-pamaccess"])
+            client.sssd.start()
+
+            # Check the results
+            assert client.auth.ssh.password("user-1", "Secret123")
+            assert not client.auth.ssh.password("user-2", "Secret123")
+    """
+
+    def __init__(self, util: PAMUtils, file: str = "/etc/security/access.conf") -> None:
```

Comment:
You already set the file default location in PAMUtils, so you don't need to set it again here.